unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/AvatarShape/Handler/AvatarShapeComponentHandler.cs

```diff
@@ -13,26 +15,38 @@ public class AvatarShapeComponentHandler : IECSComponentHandler<PBAvatarShape>
         internal IAvatarShape avatar;
         private readonly Pool pool;
         private readonly PoolableObject poolableObject;
+        private readonly IInternalECSComponent<InternalRenderers> renderersInternalComponent;
+        private IParcelScene scene;
+        private IDCLEntity entity;
 
         private bool isAvatarInitialized = false;
 
-        public AvatarShapeComponentHandler(Pool pool)
+        public AvatarShapeComponentHandler(Pool pool, IInternalECSComponent<InternalRenderers> renderersComponent)
         {
             this.pool = pool;
-            poolableObject = pool.Get(); 
+            poolableObject = pool.Get();
+            renderersInternalComponent = renderersComponent;
             avatar = poolableObject.gameObject.GetComponent<AvatarShape>();
         }
-        
+
         public void OnComponentCreated(IParcelScene scene, IDCLEntity entity)
         {
+            this.scene = scene;
+            this.entity = entity;
+
             avatar.transform.SetParent(entity.gameObject.transform, false);
+
+            avatar.internalAvatar.OnCombinedRendererUpdate += OnCombinedRendererUpdate;
         }
 
         public void OnComponentRemoved(IParcelScene scene, IDCLEntity entity)
         {
             if (avatar == null)
                 return;
-            
+
+            avatar.internalAvatar.OnCombinedRendererUpdate -= OnCombinedRendererUpdate;
+            renderersInternalComponent.RemoveFor(scene, entity);
```

Comment:
we can't just remove renderersInternalComponent for this entity, since an entity might have other renderers attached to it (gltf, meshrenderer, textshape, etc)